### PR TITLE
Added test scenarios for foreman-maintain

### PR DESCRIPTION
**Pull request by ntkathole**

### CODE REVIEW

File: upgrade_tests/test_scenarios/test_foreman_maintain.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from upgrade_tests import post_upgrade, pre_upgrade
+
+
+class Scenario_foreman_maintain(TestCase):
+    """The test class contains pre-upgrade and post-upgrade scenarios to test
+    foreman-maintain utility
+
+    Test Steps:
+
+    1. Before Satellite upgrade, Perform tests for foreman-maintain.
```
Comment (jyejare) [inline upgrade_tests/test_scenarios/test_foreman_maintain.py]: Mention what tests you will perform. like list versions.

File: upgrade_tests/test_scenarios/test_foreman_maintain.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    def test_pre_foreman_maintain_upgrade_list_versions(self):
+        """Pre-upgrade scenario that tests list of versions
+        satellite can upgrade.
+
+        :id: preupgrade-2d92ce08-0243-4f10-b4d0-b9e8a12f1dbf
+
+        :steps:
+            1. Run foreman-maintain upgrade list-versions.
+
+        :expectedresults: Versions should be 6.2.z and 6.3
```
Comment (jyejare) [inline upgrade_tests/test_scenarios/test_foreman_maintain.py]: Not hard and fast versions should be given. Just say Zstream and next version.

File: upgrade_tests/test_scenarios/test_foreman_maintain.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    def test_post_foreman_maintain_upgrade_list_versions(self):
+        """Post-upgrade scenario that tests list of versions
+        satellite can upgrade.
+
+        :id: postupgrade-2d92ce08-0243-4f10-b4d0-b9e8a12f1dbf
+
+        :steps:
+            1. Run foreman-maintain upgrade list-versions.
+
+        :expectedresults: Versions should be 6.3.z.
```
Comment (jyejare) [inline upgrade_tests/test_scenarios/test_foreman_maintain.py]: same as above

**Comment by ntkathole** [comment]
@jyejare comment addressed! Thanks.

**Pull request merged**